webhooks/main.py

```diff
@@ -45,9 +58,17 @@ async def wallet_root(
 async def topic_root(
     topic, request: Request, service: Service = Depends(Provide[Container.service])
 ):
-    payload = pformat(await request.json())
-    await service.add_topic_entry(str(topic), str(payload))
+    payload = await request.json()
+    wallet_id = {"wallet_id": (request.headers)["x-wallet-id"]}
+    payload.update(wallet_id)
+    payload = pformat(payload)
+    # redistribute by topic
     await endpoint.publish(topics=[topic], data=payload)
+    # Add data to redis
+    await service.add_topic_entry(str(topic), str(payload))
+    # redistribute per wallet
+    wallet_data = service.get_all_for_topic_by_wallet_id
```

Comment:
Shouldn't this method be called?

Reply:
probably yes. let me have a look - intrigued I accidentally deleted that with undo or sth